measurements/api/measurements.py

```diff
@@ -420,171 +390,226 @@ def list_measurements(
-            )
-        )
-    if failure is False:
-        q = q.filter(
-            and_(
-                Measurement.exc == None,
-                Measurement.residual_no == None,
-                c_msm_failure == False,
-            )
-        )
-
-    ## Create a query for fastpath
-
+    ## Create fastpath columns for query
     fpcols = [
-        func.coalesce(0).label("m_input_no"),
+        # func.coalesce(0).label("m_input_no"),
         # We use test_start_time here as the batch pipeline has many NULL measurement_start_times
-        Fastpath.measurement_start_time.label("test_start_time"),
-        Fastpath.measurement_start_time.label("measurement_start_time"),
-        func.concat(FASTPATH_MSM_ID_PREFIX, Fastpath.tid).label("measurement_id"),
-        func.coalesce(0).label("m_report_no"),
+        literal_column("measurement_start_time").label("test_start_time"),
+        literal_column("measurement_start_time").label("measurement_start_time"),
+        func.concat(FASTPATH_MSM_ID_PREFIX, sql.text("tid")).label("measurement_id"),
         func.coalesce(
-            (cast(cast(Fastpath.scores["blocking_general"], String), Float) > 0.5)
+            (cast(cast(sql.text("scores->>'blocking_general'"), String), Float) > 0.5)
         ).label("anomaly"),
         func.coalesce(false()).label("confirmed"),
-        func.coalesce(false()).label("msm_failure"),
-        cast(Fastpath.scores.label("scores"), String),
-        func.coalesce([0]).label("exc"),
-        func.coalesce(0).label("residual_no"),
-        Fastpath.report_id.label("report_id"),
-        Fastpath.probe_cc.label("probe_cc"),
-        Fastpath.probe_asn.label("probe_asn"),
-        Fastpath.test_name.label("test_name"),
-        func.coalesce(0).label("report_no"),
+        cast(sql.text("scores"), String).label("scores"),
+        literal_column("report_id"),
+        literal_column("probe_cc"),
+        literal_column("probe_asn"),
+        literal_column("test_name"),
+        literal_column("fastpath.input").label("input"),
     ]
-    if cte is None:
-        fpcols.append(Fastpath.input.label("input"))
-        assert len(fpcols) == len(cols)
-    else:
-        fpcols.append(Fastpath.input.label("input_cte_input"))
-        fpcols.append(func.coalesce(0).label("input_cte_input_no"))
-        assert len(fpcols) == len(cols) + 1
 
-    fpq = current_app.db_session.query(*fpcols)
+    mrwhere = []
+    fpwhere = []
+    query_params = {}
+
+    # Populate WHERE clauses and query_params dict
+
+    if failure is True:
+        mrwhere.append(sql.text("measurement.exc IS NOT NULL"))
+        mrwhere.append(sql.text("measurement.residual_no IS NOT NULL"))
+        mrwhere.append(sql.text("measurement.msm_failure IS TRUE"))
+
+    if failure is False:
+        mrwhere.append(sql.text("measurement.exc is NULL"))
+        mrwhere.append(sql.text("measurement.residual_no is NULL"))
+        mrwhere.append(sql.text("measurement.msm_failure is FALSE"))
```

Comment:
These two are not equivalent to how they were before, because they are now getting considered as an `AND` chain, while previously these were an `OR`.

I suggest you either do:
```
mrwhere.append(sql.text("measurement.exc IS NOT NULL OR measurement.residual_no IS NOT NULL OR measurement.msm_failure IS TRUE"))
```
thought I am not sure this works

Or you do
```
or_([
        sql.text("measurement.exc IS NOT NULL"),
        sql.text("measurement.residual_no IS NOT NULL"),
        sql.text("measurement.msm_failure IS TRUE")
])
```

Reply:
Thanks! Fixed in 77751d1a1f418ff6e4081a42f092a8f58f3321a9 and added tests.